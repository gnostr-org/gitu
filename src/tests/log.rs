use super::*;

fn setup_log() -> (TestContext, crate::state::State) {
    let mut ctx = TestContext::setup_clone(80, 20);
    commit(ctx.dir.path(), "this-should-be-at-the-top", "");
    commit(ctx.dir.path(), "this-should-not-be-visible", "");

    let state = ctx.init_state();
    (ctx, state)
}

#[test]
fn log_other_prompt() {
    let (mut ctx, mut state) = setup_log();

    state
        .update(
            &mut ctx.term,
            &[key('l'), key('l'), key('j'), key('l'), key('o')],
        )
        .unwrap();
    insta::assert_snapshot!(ctx.redact_buffer());
}

#[test]
fn log_other() {
    let (mut ctx, mut state) = setup_log();

    state
        .update(
            &mut ctx.term,
            &[
                key('l'),
                key('l'),
                key('j'),
                key('l'),
                key('o'),
                key_code(KeyCode::Enter),
            ],
        )
        .unwrap();
    insta::assert_snapshot!(ctx.redact_buffer());
}

#[test]
fn log_other_input() {
    let (mut ctx, mut state) = setup_log();

    state
        .update(
            &mut ctx.term,
            &[
                key('l'),
                key('o'),
                key('m'),
                key('a'),
                key('i'),
                key('n'),
                key('~'),
                key('1'),
                key_code(KeyCode::Enter),
            ],
        )
        .unwrap();
    insta::assert_snapshot!(ctx.redact_buffer());
}

#[test]
fn log_other_invalid() {
    let (mut ctx, mut state) = setup_log();

    state
        .update(
            &mut ctx.term,
            &[key('l'), key('o'), key(' '), key_code(KeyCode::Enter)],
        )
        .unwrap();
    insta::assert_snapshot!(ctx.redact_buffer());
}
