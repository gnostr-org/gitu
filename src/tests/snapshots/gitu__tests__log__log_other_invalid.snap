---
source: src/tests/log.rs
expression: ctx.redact_buffer()
---
Buffer {
    area: Rect { x: 0, y: 0, width: 80, height: 20 },
    content: [
        "▌On branch main                                                                 ",
        "▌Your branch is ahead of 'origin/main' by 2 commit.                             ",
        "                                                                                ",
        " Recent commits                                                                 ",
        " _______ main add this-should-not-be-visible                                    ",
        " _______ add this-should-be-at-the-top                                          ",
        " _______ origin/main add initial-file                                           ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "                                                                                ",
        "────────────────────────────────────────────────────────────────────────────────",
        "Failed due to: InvalidSpec                                                      ",
    ],
    styles: [
        x: 0, y: 0, fg: Blue, bg: Reset, underline: Reset, modifier: BOLD,
        x: 1, y: 0, fg: Yellow, bg: Reset, underline: Reset, modifier: BOLD,
        x: 0, y: 1, fg: Blue, bg: Reset, underline: Reset, modifier: DIM,
        x: 1, y: 1, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 3, fg: Yellow, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 4, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 4, fg: Yellow, bg: Reset, underline: Reset, modifier: NONE,
        x: 8, y: 4, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 9, y: 4, fg: Green, bg: Reset, underline: Reset, modifier: NONE,
        x: 13, y: 4, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 5, fg: Yellow, bg: Reset, underline: Reset, modifier: NONE,
        x: 8, y: 5, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 6, fg: Yellow, bg: Reset, underline: Reset, modifier: NONE,
        x: 8, y: 6, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 9, y: 6, fg: Red, bg: Reset, underline: Reset, modifier: NONE,
        x: 20, y: 6, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 18, fg: Reset, bg: Reset, underline: Reset, modifier: DIM,
        x: 0, y: 19, fg: Red, bg: Reset, underline: Reset, modifier: BOLD,
        x: 26, y: 19, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
    ]
}
